### PR TITLE
fix(auth): on-boarding, signup and welcome flows

### LINKED ISSUE #16855: Legacy Authentication needs better error handling
If you attempt to sign in using a legacy authentication method and it fails, beta will currently redirect you to `http://localhost:3000/auth/[social auth method]` and gives you a cryptic message.

profile___freecodecamp

Instead, it should redirect you back to /deprecated-signin and tell you "We couldn't find a freeCodeCamp account associated with that [social auth method] account."

### LINKED ISSUE #16945: [beta]: broken server routes for views
image

### PR DESCRIPTION
**Pull request by raisedadead**
Closes #16855 
Closes #16945

### CODE REVIEW

**Comment by raisedadead** [comment]
@Bouncey can you check and guide, if I should be adding this specificity, at other places? With me being little unfamiliar with all the client code.

**Review by Bouncey — COMMENTED** [review]
This was an idea I had during the reactification of the Profile page. I used `/u/:username` in the style of reddit. 

Concerns were raised that we could potentially break back links to profile pages from users' personal websites. 

This could be mitigated with a catch-all on the server that does a `User.doesExist` on the `req.baseUrl` and then redirect if required to the new endpoint.

That way, we can support depreciated sign-in and the 'legacy' `/:username`

/cc @QuincyLarson @BerkeleyTrue for thoughts.

**Comment by raisedadead** [comment]
Hi @Bouncey thanks for the additional info.

I think `User.doesExist` is an expensive call. 

However, currently the linked issue is potentially breaking the legacy login, of which we have over 200K accounts in production.

I also, agree that the use case of supporting back linking is needed too.

But, IMHO can that use case be deferred until shipping beta? I feel as minimal changes should be dome as possible to be able to ship in 2 weeks we have left.

PS: I guess, a quick fix could be checking the `/auth` in the react bootscript and passing it to server as it is. But that will not include any custom routes we have had in the production (randomAPIs.js).

**Comment by BerkeleyTrue** [comment]
@raisedadead while I appreciate the effort, this doesn't solve the issue but merely band-aids one of the symptoms. I'll continue the discussion in the issue.

**Comment by irockersup** [comment]
good post <a href="https://www.youtube.com/">Click here</a>

**Comment by QuincyLarson** [comment]
@raisedadead I'm closing this in favor of restructuring the URL path to something like /portfolio/[username] as mentioned on https://github.com/freeCodeCamp/freeCodeCamp/issues/16855

**Comment by raisedadead** [comment]
Hi @QuincyLarson I'll just update this PR for the new routes soon.

File: server/views/partials/navbar.jade [hunk 1]
```diff
@@ -29,7 +29,7 @@ nav.navbar.navbar-default.navbar-static-top.nav-height
                 a(href='https://www.freecodecamp.org/donate', target='_blank' rel='noopener') Donate
             if !user
                 li
-                    a(href='/signin') Sign Up
+                    a(href='/signin') Start Coding
```
Comment (Bouncey) [inline server/views/partials/navbar.jade]: I like this :+1:

**Comment by Bouncey** [comment]
Steps to QA? Just sign in-out-in-out? Shall I dust off my linkedin account?

**Comment by raisedadead** [comment]
> Steps to QA? Just sign in-out-in-out?

Yes. 

> Shall I dust off my linkedin account?

Yes, you will have to get a developer key. But please note that this is not mission critical because new users aren't allowed signup.

So, I am afraid you will not be able to replay the old login scenario.

The change is just a scope of info that we have limited to. On Production we had seen some complaints, which has since been resolved with the commit that a ported over.

**Comment by Bouncey** [comment]
@raisedadead I can't reach any page.

```sh

TypeError: Invalid attempt to destructure non-iterable instance
    at /home/stuart/freecodecamp/node_modules/babel-runtime/helpers/slicedToArray.js:48:13
    at /home/stuart/freecodecamp/server/component-passport.js:43:49
    at handleCallback (/home/stuart/freecodecamp/node_modules/mongodb/lib/utils.js:128:55)
    at Collection.aggregate (/home/stuart/freecodecamp/node_modules/mongodb/lib/collection.js:2524:10)
    at /home/stuart/freecodecamp/server/component-passport.js:40:10
    at /home/stuart/freecodecamp/node_modules/loopback-datasource-juggler/lib/dao.js:2176:5
    at /home/stuart/freecodecamp/node_modules/loopback-datasource-juggler/lib/dao.js:2111:9
    at /home/stuart/freecodecamp/node_modules/async/dist/async.js:1012:9
    at /home/stuart/freecodecamp/node_modules/async/dist/async.js:359:16
    at iteratorCallback (/home/stuart/freecodecamp/node_modules/async/dist/async.js:935:13)
    at /home/stuart/freecodecamp/node_modules/async/dist/async.js:843:16
    at /home/stuart/freecodecamp/node_modules/async/dist/async.js:1009:13
    at buildResult (/home/stuart/freecodecamp/node_modules/loopback-datasource-juggler/lib/dao.js:2077:11)
    at /home/stuart/freecodecamp/node_modules/loopback-datasource-juggler/lib/dao.js:2091:13
    at doNotify (/home/stuart/freecodecamp/node_modules/loopback-datasource-juggler/lib/observer.js:155:49)
    at doNotify (/home/stuart/freecodecamp/node_modules/loopback-datasource-juggler/lib/observer.js:155:49)
```

I pulled down your branch and did `rm -rf node_modules && npm i`

**Comment by raisedadead** [comment]
Ah... right. Let me check.. oh I did bump the dependencies. Sorry about that. I should not made that commit  on this branch.

I'll update this soon.

**Comment by raisedadead** [comment]
Hey @Bouncey I found an unrelated bug, if you click on the `show me my public profile` button, after you have updated via the form on the profile, the link stays pointed to the old one.

We should fix that.

Can you guide how to do that?

**Comment by Bouncey** [comment]
@raisedadead Are we talking about after updating the username?

**Comment by raisedadead** [comment]
@Bouncey yes, that is correct.

**Comment by Bouncey** [comment]
@raisedadead I'm still getting the same error? [comment](https://github.com/freeCodeCamp/freeCodeCamp/pull/16882#issuecomment-376130922)

**Comment by raisedadead** [comment]
Hi @Bouncey hmm I have removed all dependencies from my commit. So it could be something stale on your machine?

Try this:

1. Checkout staging and reset it at par as upstream
2. npm install
3. Launch the server.
4. Pull this PR
5. Test this PR

LTS for you local should be, if not already:

```
npm is 5.x.x
node is 8.x.x
mongodb is 3.6.x
```

**Review by Bouncey — APPROVED** [review]

**Pull request merged**